cubeviz/layout.py

```diff
@@ -449,13 +441,23 @@ def change_viewer_component(self, view_index, component_id, force=False):
 
         combo = self.get_viewer_combo(view_index)
 
-        component_index = combo.findData(component_id)
+        if isinstance(component_id, str):
+            component_index = combo.findText(component_id)
+        else:
+            component_index = combo.findData(component_id)
 
         if combo.currentIndex() == component_index and force:
             combo.currentIndexChanged.emit(component_index)
         else:
             combo.setCurrentIndex(component_index)
 
+    def display_component(self, component_id):
+        """
+        Displays data with given component ID in the active cube viewer.
+        """
+        view_index = self.cube_views.index(self._active_cube)
```

Comment:
I need to ask the testers about this actually. (I think only the first viewer should update not the active one)